docs/src/main/paradox/java/http/routing-dsl/directives/websocket-directives/handleWebSocketMessagesForOptionalProtocol.md

```diff
@@ -0,0 +1,17 @@
+# handleWebSocketMessagesForOptionalProtocol
+
+## Description
+
+Handles WebSocket requests with the given handler and rejects other requests with an
+@javadoc:[ExpectedWebSocketRequestRejection](akka.http.javadsl.server.ExpectedWebSocketRequestRejection$).
+
+If the `subprotocol` parameter is None any WebSocket request is accepted. If the `subprotocol` parameter is
+`Some(protocol)` a WebSocket request is only accepted if the list of subprotocols supported by the client (as
```

Comment:
Better refer to empty `Optional` and non empty `Optional`, rather than `None` and `Some(protocol)`, as these two concepts only exist in Scala Land